subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -1367,3 +1367,46 @@ Gradle 5.3 introduced the interface `TransformAction` for implementing artifact
 Using the registration method link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(org.gradle.api.Action)[DependencyHandler.registerTransform(Action)] for `ArtifactTransform` has been deprecated.
 Migrate your artifact transform to use `TransformAction` and use link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html#org.gradle.api.artifacts.dsl.DependencyHandler:registerTransform(java.lang.Class,%20org.gradle.api.Action)[DependencyHandler.registerTransform(Class, Action)] instead.
 See the <<artifact_transforms#sec:abm_artifact_transforms,user manual>> for more information on implementing `TransformAction`.
+
+[[gmm_modification_after_publication_populated]]
+==== Modifying GMM after a publication has been populated
+
+Altering the link:publishing_gradle_module_metadata.html[Gradle Module Metadata] (e.g. changing a component configuration variants) after a Maven or Ivy publication have been populated from their components, is now deprecated and scheduled to be disallowed in Gradle 9.0.
+
+Eager "realization" of the publication can happen if the following methods are called:
+
+====== Maven
+
+* link:{javadocPath}/org/gradle/api/publish/maven/MavenPublication.html#getArtifacts--[MavenPublication.getArtifacts()]
+
+====== Ivy
+
+* link:{javadocPath}/org/gradle/api/publish/ivy/IvyPublication.html#getArtifacts--[IvyPublication.getArtifacts()]
+* link:{javadocPath}/org/gradle/api/publish/ivy/IvyPublication.html#getConfigurations--[IvyPublication.getConfigurations()]
+* link:{javadocPath}/org/gradle/api/publish/ivy/IvyPublication.html#configurations(Action)--[IvyPublication.configurations(Action)]
+
+Before, it was possible to have this code block without any warnings, but it created inconsistencies between the published artifacts:
+
+```kotlin
+publishing {
+    publications {
+        create<MavenPublication>("maven") {
+            from(components["java"])
+        }
+        create<IvyPublication>("ivy") {
+            from(components["java"])
+        }
+    }
+}
+
+// These calls eagerly populate the Maven and Ivy publications
+(publishing.publications["maven"] as MavenPublication).artifacts
+(publishing.publications["ivy"] as IvyPublication).artifacts
+
+(components["java"] as DefaultAdhocSoftwareComponent).apply {
+  withVariantsFromConfiguration(configurations["apiElements"]) { skip() }
+  withVariantsFromConfiguration(configurations["runtimeElements"]) { skip() }
+}
+```
+
+In this example, the Maven and Ivy publications will contain the main JAR artifacts for the project, whereas the GMM link:https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/design/gradle-module-metadata-latest-specification.md[module file] will have them skipped.
```

Comment:
```suggestion
In this example, the Maven and Ivy publications contain the main JAR artifacts for the project, whereas the Gradle Module Metadata (GMM) link:https://github.com/gradle/gradle/blob/master/subprojects/docs/src/docs/design/gradle-module-metadata-latest-specification.md[module file] will omit them.
```
Suggestion: switch to "omit" since "skipped" doesn't quite capture the full consequences here (as far as I can tell; I *think* we're talking about missing artifacts, no?). Define "GMM", which many users might not be aware of already.

Reply:
Thanks @nathan-contino. 

About this change, I was omitting it because the GMM is already described on the first sentence of the section. 

The idea in the second part of the sentence was to make reference to the `skip()` method call, but seems to me a good suggestion.